egs/librimix/ConvTasNet/eval.py

```diff
@@ -0,0 +1,149 @@
+import argparse
+import json
+import os
+import random
+from pprint import pprint
+import numpy as np
+import pandas as pd
+import soundfile as sf
+import torch
+import yaml
+from pb_bss.evaluation import InputMetrics, OutputMetrics
+from tqdm import tqdm
+
+from asteroid.data.librimix_dataset import LibriMix
+from asteroid.losses import PITLossWrapper, pairwise_neg_sisdr
+from asteroid.torch_utils import load_state_dict_in
+from asteroid.utils import tensors_to_device, average_arrays_in_dic
+from model import make_model_and_optimizer
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--test_dir', type=str, required=True,
+                    help='Test directory including the json files')
+parser.add_argument('--use_gpu', type=int, default=0,
+                    help='Whether to use the GPU for model execution')
+parser.add_argument('--exp_dir', default='exp/tmp',
+                    help='Experiment root')
+parser.add_argument('--n_save_ex', type=int, default=10,
+                    help='Number of audio examples to save, -1 means all')
+
+compute_metrics = ['si_sdr', 'mir_eval_sdr', 'mir_eval_sir', 'mir_eval_sar',
+                   'stoi']
+
+
+def main(conf):
+    # Make the model
+    model, _ = make_model_and_optimizer(conf['train_conf'])
+    # Load best model
+    with open(os.path.join(conf['exp_dir'], 'best_k_models.json'), "r") as f:
+        best_k = json.load(f)
+    best_model_path = min(best_k, key=best_k.get)
+    # Load checkpoint
+    checkpoint = torch.load(best_model_path, map_location='cpu')
+    state = checkpoint['state_dict']
+    state_copy = state.copy()
+    # Remove unwanted keys
+    for keys, values in state.items():
+        if keys.startswith('loss'):
+            del state_copy[keys]
+            print(keys)
+    model = load_state_dict_in(state_copy, model)
+
+    # Handle device placement
+    if conf['use_gpu']:
+        model.cuda()
+    model_device = next(model.parameters()).device
+
+    test_set = LibriMix(conf['test_dir'], None,
+                        conf['sample_rate'],
+                        conf['train_conf']['data']['n_src'])
```

Comment:
Can you add the keywords in the call please? It makes it much easier to read. In particular, it's not obvious that `None` means, use all segment length, which is crucial !

Reply:
Sure and you made me realize that my arguments don't match the dataloader arguments ...